seacrowd/sea_datasets/xstorycloze/xstorycloze.py

```diff
@@ -0,0 +1,188 @@
+                    December 7-11, 2022},
+    pages        = {9019--9052},
+    publisher    = {Association for Computational Linguistics},
+    year         = {2022},
+    url          = {https://doi.org/10.18653/v1/2022.emnlp-main.616},
+    doi          = {10.18653/V1/2022.EMNLP-MAIN.616},
+}
+"""
+
+_DATASETNAME = "xstorycloze"
+_DESCRIPTION = """\
+XStoryCloze consists of the professionally translated version of the English StoryCloze
+dataset (Spring 2016 version) to 10 non-English languages. This dataset is released by
+Meta AI.
+"""
+_HOMEPAGE = "https://huggingface.co/datasets/juletxara/xstory_cloze"
+_LICENSE = Licenses.CC_BY_SA_4_0.value
+
+_LOCAL = False
+_BASE_URL = "https://huggingface.co/datasets/juletxara/xstory_cloze/resolve/main/spring2016.val.{lang}.tsv.split_20_80_{split}.tsv"
+_SUPPORTED_TASKS = [Tasks.COMMONSENSE_REASONING, Tasks.SELF_SUPERVISED_PRETRAINING]
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class xStoryClozeDataset(datasets.GeneratorBasedBuilder):
+    """XStoryCloze subset for Indonesian and Burmese language."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = [TASK_TO_SCHEMA[task].lower() for task in _SUPPORTED_TASKS]
+    SEACROWD_SUBSET = ["id", "my"]
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{subset}_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description=f"{_DATASETNAME} {subset} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}_{subset}",
+        ) 
+        for subset in SEACROWD_SUBSET
+    ] + [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{subset}_seacrowd_{schema}",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} {subset} SEACrowd schema",
+            schema=f"seacrowd_{schema}",
+            subset_id=f"{_DATASETNAME}_{subset}",
+        )
+        for subset, schema in list(itertools.product(SEACROWD_SUBSET, SEACROWD_SCHEMA_NAME))
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_{SEACROWD_SUBSET[0]}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "story_id": datasets.Value("string"),
+                    "input_sentence_1": datasets.Value("string"),
+                    "input_sentence_2": datasets.Value("string"),
+                    "input_sentence_3": datasets.Value("string"),
+                    "input_sentence_4": datasets.Value("string"),
+                    "sentence_quiz1": datasets.Value("string"),
+                    "sentence_quiz2": datasets.Value("string"),
+                    "answer_right_ending": datasets.Value("int32"),
+                }
+            )
+        elif self.config.schema == f"seacrowd_{(schema := self.SEACROWD_SCHEMA_NAME[0])}":
```

Comment:
Small nit: because you'll have two schemas, it might be better to be explicit here and just write the schema name itself. Same comment in the `_generate_examples` code.

Reply:
Hi sorry for the late reply, I implemented `_ssp` here because in the dataloader issue #69 there are two tasks and one for language modelling.

Reply:
Okay then, let's ask @SamuelCahyawijaya to review LangMod task on this dataset

Reply:
@sabilmakbar  @akhdanfadh  @ljvmiranda921 : Hi all, sorry for the late reply. We can remove the Language Modeling task from the implementation. Sorry for the confusion 🙏🏻